### ISSUE TITLE
dotnet format makes no changes, but --check reports encoding issues

### ISSUE THREAD

**Issue by owensearls** [dotnet/format#507]
After upgrading to the latest release of this package, version 3.2.107702, when I ran `dotnet format` for the first time it modified the encoding of some files. After that, running `dotnet format` makes no changes, but when I run `dotnet format --check --dry-run` it returns a non zero exit code and prints and message about the encoding of almost every file in my project.

```
MyProject/Whatever.cs(1,1): Fix file encoding.
```

When I run with or without the check flags, it prints a message saying it formatted every file in my project, which is also new, but nothing has changed, at least that git can detect.

```
Formatted code file 'Whatever.cs'.
```

I am using the editorconfig option `charset = utf-8`, and have observed this issue on MacOS and linux.

**Comment by jmarolf** [comment]
@JoeRobich can you reproduce this in master?

**Comment by JoeRobich** [comment]
Interesting, It could be a bug with the MSBuildProjectLoader.

When using the folder option (`-f`), I do not get subsequent formatted code messages after my initial format. However, If I use the workspace option (`-w`), I see all files get reported as formatted.

**Comment by JoeRobich** [comment]
The issue may be here - http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces.MSBuild/MSBuild/MSBuildProjectLoader.Worker.cs,0cc606639ff7cd3d,references

It is loading the DocumentInfo with the Encoding of the CommandLineArgs, whatever that means, instead of the encoding of the underlying file itself.

**Comment by JoeRobich** [comment]
From the comment, it seems like maybe `null` should just be passed in to allow the FileTextLoader to sort it out. http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/FileTextLoader.cs,56

**Comment by JoeRobich** [comment]
@owensearls I also wanted to check whether you have a CodePage environment variable set and its value. `echo $CodePage`

**Comment by owensearls** [comment]
@JoeRobich I don't have that environment variable set. Let me know if you want me to try anything, I could't find anything about what it does on google.

Thanks for your quick response!

**Comment by schmitch** [comment]
@JoeRobich will there be a new 3.2 release soon? I accidently installed the latest `dotnet-format` on our ci servers and needed to downgrade the tool afterwards since we use `dotnet-format --check` their, which failed because of this.

**Comment by JoeRobich** [comment]
@schmitch I just published 3.3.111304 to NuGet. Thanks for using dotnet-format!
